include/laser_filters/scan_shadows_filter.h

```diff
@@ -111,13 +124,33 @@ class ScanShadowsFilter : public filters::FilterBase<sensor_msgs::LaserScan>
       ROS_ERROR("max_angle must be max_angle <= 180. Forcing max_angle = 180.\n");
       max_angle_ = 180.0;
     }
+
     shadow_detector_.configure(
         angles::from_degrees(min_angle_),
         angles::from_degrees(max_angle_));
 
+    param_config.min_angle = min_angle_;
+    param_config.max_angle = max_angle_;
+    param_config.window = window_;
+    param_config.neighbors = neighbors_;
+    param_config.remove_shadow_start_point = remove_shadow_start_point_;
+    dyn_server_->updateConfig(param_config);
+
     return true;
   }
 
+  void reconfigureCB(ScanShadowsFilterConfig& config, uint32_t level)
+  {
```

Comment:
You should also grab `own_mutex_` here, right?

Reply:
This is being handled by the reconfigure server as far as I can see: https://github.com/ros/dynamic_reconfigure/blob/melodic-devel/include/dynamic_reconfigure/server.h#L216

Reply:
Good point! I hadn't realized it used a mutex that was passed in to it.